pkg/cmd/gist/view/view.go

```diff
@@ -93,12 +94,22 @@ func viewRun(opts *ViewOptions) error {
 
 	theme := opts.IO.DetectTerminalTheme()
 	markdownStyle := markdown.GetStyle(theme)
-	if err := opts.IO.StartPager(); err != nil {
-		fmt.Fprintf(opts.IO.ErrOut, "starting pager failed: %v\n", err)
-	}
-	defer opts.IO.StopPager()
 
 	render := func(gf *shared.GistFile) error {
+		isBinary, err := shared.DetectedBinaryFileContents(gf.Content, true)
+		if err != nil {
+			return err
+		}
+		if isBinary {
+			var header [128]byte
+			_, err := io.ReadFull(strings.NewReader(gf.Content), header[:])
+			if err != nil {
+				return err
+			}
+
+			gf.Content = string(header[:]) + "\n\nSkipping rendering binary content..."
```

Comment:
Let's not render even a part of the binary file, since raw bytes could mess up terminal output. Just a notice will do.

Reply:
Got it.